src/AnnotationThread.js

```diff
@@ -206,46 +199,64 @@ class AnnotationThread extends EventEmitter {
      * @param {string} message - Text of annotation to save
      * @return {Promise} - Annotation create promise
      */
-    saveAnnotation = (type, message) => {
+    saveAnnotation(type, message) {
         const annotationData = this.createAnnotationData(type, message);
 
         // Save annotation on client
-        const tempAnnotationID = AnnotationAPI.generateID();
-        const tempAnnotation = {
-            id: tempAnnotationID,
+        const id = AnnotationAPI.generateID();
+        this.updateAnnotationThread({
+            id,
+            message,
             permissions: {
                 can_edit: true,
                 can_delete: true
             },
-            message,
-            location: this.location,
-            created_at: new Date().toLocaleString(),
-            created_by: this.api.user,
+            createdBy: this.api.user,
+            createdAt: new Date().toLocaleString(),
             isPending: true
-        };
-        this.annotations.push(tempAnnotation);
+        });
+
         this.renderAnnotationPopover();
 
         this.state = STATES.inactive;
 
         // Save annotation on server
         return this.api
             .create(annotationData)
-            .then((savedAnnotation) => this.updateTemporaryAnnotation(tempAnnotation, savedAnnotation))
-            .catch((error) => this.handleThreadSaveError(error, tempAnnotationID));
-    };
+            .then((savedAnnotation) => this.updateTemporaryAnnotation(id, savedAnnotation))
+            .catch((error) => this.handleThreadSaveError(error, id));
+    }
+
+    updateAnnotationThread(data) {
+        if (!this.threadNumber) {
+            this.id = data.id;
+            this.threadNumber = data.threadNumber;
+            this.threadID = data.threadID;
+        }
+
+        // If annotation is the first in the thread
+        const { message } = data;
+        if (message && message.trim() !== '') {
```

Comment:
if the annotation is just created and has a message, does that mean that lines 242-244 (updating createdAt, createdBy, modifiedAt) fields don't get updated?